src/components/LoadingSpinner/LoadingSpinner.tsx

```diff
@@ -1,11 +1,47 @@
 import { cn } from '@/lib/utils'
-import Lottie from 'lottie-react'
-import loadingAnimation from '@/public/loading.json'
-
-const LoadingSpinner = ({ className = '' }) => (
-  <div className={cn('flex justify-center', className)}>
-    <Lottie animationData={loadingAnimation} className="w-1/2" loop={true} />
-  </div>
-)
+import { SpinnerSize } from '.'
+
+type LoadingSpinnerProps = {
+  className?: string
+  size?: SpinnerSize
+}
+
+const LoadingSpinner = ({ className = '', size = 'medium' }: LoadingSpinnerProps) => {
+  // Define size classes based on the size prop
+  let sizeClass = ''
+
+  if (typeof size === 'number') {
```

Comment:
Instead of having sizes, is there any way of having this component responsive depending on the parent size?